src/Symfony/Component/HttpClient/Test/Retry/NoRetryStrategy.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Test\Retry;
+
+use Symfony\Component\HttpClient\Response\AsyncContext;
+use Symfony\Component\HttpClient\Retry\RetryStrategyInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+
+final class NoRetryStrategy implements RetryStrategyInterface
+{
+    public function shouldRetry(AsyncContext $context, ?string $responseContent, ?TransportExceptionInterface $exception): ?bool
+    {
+        return false;
+    }
+
+    public function getDelay(AsyncContext $context, ?string $responseContent, ?TransportExceptionInterface $exception): int
+    {
+        throw new \LogicException();
```

Comment:
Maybe it would be worth it to pass also context and exception to this one?

Reply:
This will never be called since `shouldRetry()` returns false.

Reply:
Then why return 0 and add a comment it will not happen?